Regular-Question-Answers/path_with_maximum_probability.cpp

```diff
@@ -0,0 +1,145 @@
+// Author: Novruz Amirov
+// Question Link: https://leetcode.com/problems/path-with-maximum-probability/
+
+/*
+    Problem Description:
+    
+        We are given an UNDIRECTED graph of n nodes.
+        The nodes are 0-indexed, which means in adjacency
+        list we do not need to use 1 additional memory for
+        1-index nodes.
+        Then, we are given succProb vector, which displays
+        us the propability of the edge between 2 nodes.
+        Finally, we are given start and end inputs to
+        know from which node to start and end.
+        
+        Note: The probability is calculated by multiplying.
+        
+        Example:
+            The Graph:
+                (0) ---> (1), with the weight of 0.8
+                (0) ---> (2), weight = 0.6
+                (1) ---> (3), weight = 0.1
+                (2) ---> (3), weight = 0.9
+                
+            starting point is 0, ending point is 3.
+            Totally there is 2 way of coming to the node 3
+            starting from the node 1. 
+            First is (0) ---> (1) ---> (3), probability = 0.08
+            Second is (0) ---> (2) ---> (3), probability = 0.54
+        
+            Because 0.54 is bigger than 0.08, we will return 0.54
+        
+    
+    Data types, structurem, and algortihms used for this question:
+        We will use Data Types of bool, int, and double.
+            bool -> for visited vector to check if the node is visited or not.
+            int -> for the node numbers.
+            double -> for the probability.
+        
+        As a Data Structures:
+            vector<vector<pair<int, double>>> for adjacency list,
+            adjacency list is more efficient than adjacency matrix
+            when used in Dijkstra algorithm.
+            
+            We will also use vectors as a data structure for
+            visited and probabilities array, to check if the
+            node is visited or not, and to store the probabilities
+            of each node,
+            
+            Priority Queue will be used for Dijkstra algorithm.
+        
+        As an Algorithm:
+            As I mention above, we will use Dijkstra algorithm, 
+            normally Dijsktra is used for finding the minimum distance
+            or probability, but we will use it reversely for maximum.
+        
+*/
+
+
+
+
+class Solution {
+public:
+    
+    double maxProbability(int n, vector<vector<int>>& edges, vector<double>& succProb, int start, int end) {
+        
+        // initializing 0-index adjacency list to use the Graph.
+        vector< vector< pair<int, double> > >adjacency_list(n, vector< pair<int, double> >(0)); 
```

Comment:
`vector<vector<pair<int, double>>> adjacencyList(n, vector<pair<int, double>>(0)); `
do not forget to use camel case
also the spaces are corrected